windows_install_build_dependencies.bat

```diff
@@ -270,52 +276,92 @@ IF /I EXIST %python_path% (
         :: Install python
         curl -k %python_url% -o %opt_install_dir%\%python_full_name%.exe
         if !errorlevel! neq 0 exit /b !errorlevel!
-        %opt_install_dir%\%python_full_name%.exe /quiet /uninstall
-        if !errorlevel! neq 0 exit /b !errorlevel!
-        %opt_install_dir%\%python_full_name%.exe /passive /quiet /simple Include_symbols=1 TargetDir=%python_path%
-        if !errorlevel! neq 0 exit /b !errorlevel!
-        %python_path%\python.exe -m ensurepip --upgrade
-        if !errorlevel! neq 0 exit /b !errorlevel!
-        :: setuptools<60.0 required for numpy1.23 on python311 to install
-        %python_path%\python.exe -m pip install "setuptools<60.0" "numpy==1.23" "Jinja2==3.1.5" "MarkupSafe==3.0.2"
+        call :UninstallPython
         if !errorlevel! neq 0 exit /b !errorlevel!
+        call :InstallPython
+        IF /I not EXIST %python_path%\python.exe (
+            echo "Python installation failed. Errorlevel: !errorlevel!"
+            echo "To fix the installation run %opt_install_dir%\%python_full_name%.exe in GUI and press repair."
+            echo "Please review the installation log: python/install.log"
+            echo "Rerun the windows_install_build_dependencies.bat, once the installation is fixed."
+            exit /b !errorlevel!
+        )
     ) ELSE (
         echo [INFO] ::::::::::::::::::::::: %python_path% already installed
     )
 ) ELSE (
```

Comment:
why python is uninstalled when present? It looks like it will be reinstalled with every execution.
maybe just the version should be checked ?

Reply:
The uninstall help with clean installation. Its only done once, as if python.exe is present it will no longer run.

Reply:
The check is if the install_python.exe is present.